va/validation-authority.go

```diff
@@ -525,31 +479,25 @@ func (va *ValidationAuthorityImpl) validate(authz core.Authorization, challengeI
 		Requester:   authz.RegistrationID,
 		RequestTime: va.clk.Now(),
 	}
-	if !authz.Challenges[challengeIndex].IsSane(true) {
-		chall := &authz.Challenges[challengeIndex]
-		chall.Status = core.StatusInvalid
-		chall.Error = &probs.ProblemDetails{Type: probs.MalformedProblem,
-			Detail: fmt.Sprintf("Challenge failed sanity check.")}
-		logEvent.Challenge = *chall
-		logEvent.Error = chall.Error.Detail
-	} else {
-		var err error
+	challenge := &authz.Challenges[challengeIndex]
+	vStart := va.clk.Now()
+	validationRecords, prob := va.validateChallengeAndCAA(authz.Identifier, *challenge, authz.RegistrationID)
+	va.stats.TimingDuration(fmt.Sprintf("VA.Validations.%s.%s", challenge.Type, challenge.Status), time.Since(vStart), 1.0)
 
-		vStart := va.clk.Now()
-		authz.Challenges[challengeIndex], err = va.validateChallengeAndCAA(authz.Identifier, authz.Challenges[challengeIndex], authz.RegistrationID)
-		va.stats.TimingDuration(fmt.Sprintf("VA.Validations.%s.%s", authz.Challenges[challengeIndex].Type, authz.Challenges[challengeIndex].Status), time.Since(vStart), 1.0)
-
-		if err != nil {
-			logEvent.Error = err.Error()
-		} else if !authz.Challenges[challengeIndex].RecordsSane() {
-			chall := &authz.Challenges[challengeIndex]
-			chall.Status = core.StatusInvalid
-			chall.Error = &probs.ProblemDetails{Type: probs.ServerInternalProblem,
-				Detail: "Records for validation failed sanity check"}
-			logEvent.Error = chall.Error.Detail
-		}
-		logEvent.Challenge = authz.Challenges[challengeIndex]
+	challenge.ValidationRecord = validationRecords
+	if prob != nil {
+		challenge.Status = core.StatusInvalid
+		challenge.Error = prob
+		logEvent.Error = prob.Error()
+	} else if !authz.Challenges[challengeIndex].RecordsSane() {
```

Comment:
Sigh, at some point we gotta fix this API (which we are all aware of), because I can never remember how RecordsSane is different than IsSane(). Not for this PR, of course